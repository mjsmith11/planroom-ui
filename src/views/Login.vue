<template>
    <div class="container">
        <div class="col-xs-10 col-sm-8 col-md-6 offset-md-3 offset-sm-2 offset-xs-1 column">
            <div class="card">
                <h4 class="card-header">Login</h4>
                <div class="card-body">
                    <div class="alert alert-danger" role="alert" v-if="loginFailure">Login Failed</div>
                    <form @submit.prevent = "login">
                        <div class="form-group">
                            <input type="email"
                                class="form-control"
                                id="email"
                                placeholder="Email"
                                maxlength="50"
                                v-model="email"
                                required
                            >
                        </div>
                        <div class="form-group">
                            <input type="password"
                                class="form-control"
                                id="password"
                                placeholder="Password"
                                maxlength="50"
                                v-model="password"
                                required
                            >
                        </div>
                        <button class="btn btn-outline-success" type="submit">Login</button>
                    </form>
                </div>
            </div>
        </div>
    </div>
</template>
<script>
export default {
  data () {
    return {
      email: '',
      password: '',
      loginFailure: false
    }
  },
  methods: {
    login () {
      this.loginFailure = false
      let email = this.email
      let password = this.password
      this.$store.dispatch('login', { email, password })
        .then(() => this.$router.push('/jobs'))
        .catch(() => {
          this.loginFailure = true
        })
    }
  }

}
</script>

<style scoped>
  div.column {
    padding-top: 30px;
  }
button {
    width: 100%;
  }
  .card {
      box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
  }
</style>
