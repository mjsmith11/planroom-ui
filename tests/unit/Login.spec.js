import { mount } from '@vue/test-utils'
import Login from '@/views/Login'
import flushPromises from 'flush-promises'

describe('App', () => {
  it('has expected html', () => {
    const cmp = mount(Login, {

    })
    expect(cmp.html()).toMatchSnapshot()
  })

  it('handles successful login', async () => {
    const $router = {
      push: jest.fn()
    }
    const $store = {
      dispatch: jest.fn().mockReturnValue(new Promise((resolve, reject) => {
        resolve()
      }))
    }
    const cmp = mount(Login, {
      mocks: {
        $router,
        $store
      }
    })
    const emailField = cmp.find('#email')
    emailField.setValue('test@test.com')
    const passwordField = cmp.find('#password')
    passwordField.setValue('myPassword')

    cmp.find('form').trigger('submit')
    expect($store.dispatch).toBeCalledWith('login', { email: 'test@test.com', password: 'myPassword' })

    await flushPromises()

    expect($router.push).toHaveBeenCalledWith('/jobs')
  })

  it('handles failed login', async () => {
    const $router = {
      push: jest.fn()
    }
    const $store = {
      dispatch: jest.fn().mockReturnValue(new Promise((resolve, reject) => {
        reject(new Error('Login Failed'))
      }))
    }
    const cmp = mount(Login, {
      mocks: {
        $router,
        $store
      }
    })
    const emailField = cmp.find('#email')
    emailField.setValue('test@test.com')
    const passwordField = cmp.find('#password')
    passwordField.setValue('wrongPassword')

    cmp.find('form').trigger('submit')
    expect($store.dispatch).toBeCalledWith('login', { email: 'test@test.com', password: 'wrongPassword' })

    await flushPromises()

    expect($router.push).not.toHaveBeenCalled()
    const alert = cmp.find('div.alert')
    expect(alert.classes()).toContain('alert-danger')
    expect(alert.html()).toContain('Login Failed')
  })
})
