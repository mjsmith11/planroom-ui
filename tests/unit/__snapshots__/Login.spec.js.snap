// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has expected html 1`] = `
<div class="container">
  <div class="col-xs-10 col-sm-8 col-md-6 offset-md-3 offset-sm-2 offset-xs-1 column">
    <div class="card">
      <h4 class="card-header">Login</h4>
      <div class="card-body">
        <!---->
        <form>
          <div class="form-group">
            <input type="email" id="email" placeholder="Email" maxlength="50" required="required" class="form-control">
          </div>
          <div class="form-group">
            <input type="password" id="password" placeholder="Password" maxlength="50" required="required" class="form-control">
          </div>
          <button type="submit" class="btn btn-outline-success">Login</button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
